R/get_truth.R

```diff
@@ -221,4 +221,76 @@ preprocess_usafacts <- function (save_location="./data-truth/usafacts/"){
       dplyr::rename(value = inc) %>%
       dplyr::select(date, location, value),
     path = paste0(save_location,"truth_usafacts-Incident Cases.csv"))
+}
+
+
+#' Preprocess raw truth data from JHU CSSE into Cumulative/Incident - Deaths/Cases and write to CSVs
+#' 
+#' @param save_location character specifying the location of to save raw truth data.
+#' Default to "./data-truth/"
+#' 
+#' @export
+preprocess_jhu <- function (save_location="./data-truth/"){
+  # Location name
+  location_names <- covidData::fips_codes[, c("location", "location_name")]
+  
+  # National data
+  national_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  national_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+
+  # State data
+  state_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  state_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # County data
+  county_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  county_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # Combine to get cases data
+  cases_dframes <- dplyr::bind_rows(national_cases_dframes, state_cases_dframes, county_cases_dframes)
```

Comment:
We could simplify this code by just providing `spatial_resolution = c("county", "state", "national")` in one call to `load_jhu_data` for cases and again in a second call for deaths. This would make things a little simpler by reducing the number of calls and eliminating the calls to `dplyr::bind_rows` here and on line 268.

Reply:
This is great to know! I didn't realize that it mentioned you can input a [vector ](https://github.com/reichlab/covidData/blob/master/R/jhu_data.R#L6) of spatial resolution. I have updated the code accordingly